ecl/hql/hqlgram.y

```diff
@@ -8812,6 +8812,10 @@ simpleDataSet
 
                             bool hasFileposition = getBoolAttributeInList(extra, filepositionAtom, true);
                             record.setown(parser->checkIndexRecord(record, $5, extra));
+
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
+                                extra.setown(createComma(extra.getClear(), createAttribute(_disallowed_Atom)));
```

Comment:
@jakesmith @richardkchapman Is there agreement that (although there is other ways to get this done) this is good enough?